lib/internal/mime.js

```diff
@@ -182,103 +233,82 @@ class MIMEParams {
     for (const { 0: key, 1: value } of this.#data) {
       const encoded = encode(value);
       // Ensure they are separated
-      if (ret.length) ret += ';';
+      if (ret.length > 0) ret += ';';
       ret += `${key}=${encoded}`;
     }
+
     return ret;
   }
 
   // Used to act as a friendly class to stringifying stuff
   // not meant to be exposed to users, could inject invalid values
   static parseParametersString(str, position, params) {
     const paramsMap = params.#data;
-    const endOfSource = SafeStringPrototypeSearch(
-      StringPrototypeSlice(str, position),
-      START_ENDING_WHITESPACE,
-    ) + position;
-    while (position < endOfSource) {
+    const endOfSource = str.length;
+    const marker = { position };
+    while (marker.position < endOfSource) {
       // Skip any whitespace before parameter
-      position += SafeStringPrototypeSearch(
-        StringPrototypeSlice(str, position),
-        END_BEGINNING_WHITESPACE,
-      );
-      // Read until ';' or '='
-      const afterParameterName = SafeStringPrototypeSearch(
-        StringPrototypeSlice(str, position),
-        EQUALS_SEMICOLON_OR_END,
-      ) + position;
-      const parameterString = toASCIILower(
-        StringPrototypeSlice(str, position, afterParameterName),
-      );
-      position = afterParameterName;
-      // If we found a terminating character
-      if (position < endOfSource) {
-        // Safe to use because we never do special actions for surrogate pairs
-        const char = StringPrototypeCharAt(str, position);
+      moveUntil(str, isNotWhiteSpace, marker);
+      const parameterName = moveUntil(str, isSemicolonOrEquals, marker, { toASCII: true });
+      // If we are at end of the string, it cannot have a value
+      if (marker.position > endOfSource) break;
```

Comment:
Can we remove this extra branch somehow? Each branch reduces performance

Reply:
Simplified to pure `if`s, so far show slight improvement (1 - 5%).
Pushed them, but remain the same.

Sadly cannot be easily bypassed, as the checks do two different things, checking whether we run out of bounds, and checking whether the parameters have value or not